Source/Core/Core/HW/DSP.cpp

```diff
@@ -231,30 +231,51 @@ void Shutdown()
 
 void RegisterMMIO(MMIO::Mapping* mmio, u32 base)
 {
+  constexpr u16 WMASK_NONE = 0x0000;
+  constexpr u16 WMASK_AR_INFO = 0X007f;
+  constexpr u16 WMASK_AR_REFRESH = 0X07ff;
+  constexpr u16 WMASK_AR_HI_RESTRICT = 0x03ff;
+  constexpr u16 WMASK_AR_CNT_DIR_BIT = 0x8000;
+  constexpr u16 WMASK_LO_ALIGN_32BIT = 0xffe0;
+  const u16 WMASK_AUDIO_HI_RESTRICT = SConfig::GetInstance().bWii ? 0x1fff : 0x03ff;
```

Comment:
Does this work as intended if you launch a GameCube game from the Wii menu? (I'm not sure which behavior is actually intended)

Reply:
The MMIO handlers are only registered once, so the Wii behavior will be inherited in that case. If that's wrong, then CommandProcessor::RegisterMMIO() is likely wrong as well.

Reply:
Good catch. I have now verified that a Wii in GC mode behaves like a real GC. I've deferred the mask selection to time of use to properly support this scenario.

Someone should audit the other uses of bWii throughout Dolphin. The usage in CommandProcessor, at least, is very likely incorrect for this case...